### DISCUSSION TITLE
Create PRs for framework compatible packages only

### DISCUSSION THREAD

**Discussion by dmiszczak** [renovatebot/renovate#32738] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Azure DevOps 

### Please tell us more about your question or problem

We are using Mend Renovate in the organization mostly to create PRs for .NET and Java apps. However at the moment we need to weed out the PRs with the library updates that are not compatible with the framework version we are running the app on. Just a recent example: we have an app written in .NET 8 but we are getting a PR for a Microsoft.AspNetCore.Mvc.NewtonsoftJson version 9.0.0 which is compatible only with .NET 9

**My question is**, is there a way of setting up configuration so the PRs created will be only compatible with a specific version  of the framework ? or better yet the Renovate will recognize the underlying framework and only creates the PRs with updates compatible to it.

I hope that makes sense :) 

Thanks in advance ! 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Renovate has similar functionality already, but it doesn't seem to be supported for the `nuget` datasource. See https://docs.renovatebot.com/language-constraints-and-upgrading/ for some more information.

Such a solution would require:
- .NET framework information to be available from the nuget datasource, and
- .NET framework compatibility to be in a standardized location in a project (ideal) or user to configure `constraints.nuget` in each repo
